docs/csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer.md

```diff
@@ -1,33 +1,31 @@
 ---
 title: "How to: Initialize Objects by Using an Object Initializer - C# Programming Guide"
 ms.custom: seodec18
-ms.date: 07/20/2015
+ms.date: 12/20/2018
 helpviewer_keywords: 
   - "object initializers [C#], how to use"
   - "objects [C#], initializing"
 ms.assetid: 4b75ebb2-2e29-43de-929c-d736a8f27ce6
 ---
 # How to: Initialize Objects by Using an Object Initializer (C# Programming Guide)
+
 You can use object initializers to initialize type objects in a declarative manner without explicitly invoking a constructor for the type.  
   
- The following examples show how to use object initializers with named objects. The compiler processes object initializers by first accessing the default instance constructor and then processing the member initializations. Therefore, if the default constructor is declared as `private` in the class, object initializers that require public access will fail.  
+The following examples show how to use object initializers with named objects. The compiler processes object initializers by first accessing the default instance constructor and then processing the member initializations. Therefore, if the default constructor is declared as `private` in the class, object initializers that require public access will fail.
   
- You must use an object initializer if you're defining an anonymous type. For more information, see [How to: Return Subsets of Element Properties in a Query](../../../csharp/programming-guide/classes-and-structs/how-to-return-subsets-of-element-properties-in-a-query.md).  
+You must use an object initializer if you're defining an anonymous type. For more information, see [How to: Return Subsets of Element Properties in a Query](how-to-return-subsets-of-element-properties-in-a-query.md).  
   
 ## Example  
- The following example shows how to initialize a new `StudentName` type by using object initializers.  
-  
- [!code-csharp[csProgGuideLINQ#35](../../../csharp/programming-guide/arrays/codesnippet/CSharp/how-to-initialize-objects-by-using-an-object-initializer_1.cs)]  
-  
-## Example  
- The following example shows how to initialize a collection of `StudentName` types by using a collection initializer. Note that a collection initializer is a series of comma-separated object initializers.  
-  
- [!code-csharp[csProgGuideLINQ#36](../../../csharp/programming-guide/arrays/codesnippet/CSharp/how-to-initialize-objects-by-using-an-object-initializer_2.cs)]  
-  
-## Compiling the Code  
- To run this code, copy and paste the class into a Visual C# console application project that has been created in Visual Studio.  
+
+The following example shows how to initialize a new `StudentName` type by using object initializers. This example sets properties in the `StudentName` type:
   
+[!code-csharp-interactive[InitializerObjectExample](../../../../samples/snippets/csharp/programming-guide/classes-and-structs/object-collection-initializers/HowToObjectInitializers.cs#HowToObjectInitializers)]  
```

Comment:
This doesn't run automatically because `Main()` is not in a class. Either have to put it in a class or add a call to it directly.